aten/src/ATen/TensorCompare.h

```diff
@@ -0,0 +1,100 @@
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+#include "ATen/ExpandUtils.h"
+#include "ATen/NativeFunctions.h"
+
+
+template <template<typename> class, typename, typename, template<template<typename> class, typename, typename> class>
```

Comment:
Can you add names to those template arguments and possibly split them to have only on of them in a single line? This would serve as a good reference for what are they supposed to mean, because right now it looks like a random sequence of `template` and `typename` 😄